### DISCUSSION TITLE
Is there any pretrained model zoo/hub for common used networks in flax

### DISCUSSION THREAD

**Discussion by wztdream** [google/flax#1563] [Q&A] [answered]
Hi,
Is there any pretrained model (on Imagenet)  for common used networks like RestNet, Goolenet,vgg,etc. Similar with tensorflow or pyrtorch zoo?

It is very helpful to have this pretrained checkpoints and models as pre-training on large dataset is time and resource consuming

**Comment by marcvanzee** [comment, marked as answer]
Hi, many of our examples' pretrained weights are publicly available. Please see the README of those examples, e.g. [Imagenet](https://github.com/google/flax/tree/main/examples/imagenet), [LM1B](https://github.com/google/flax/tree/main/examples/lm1b), [WMT](https://github.com/google/flax/tree/main/examples/wmt). The Imagenet example also contains a Colab showing how the pretrained weights are loaded.

**Reply by wztdream** [reply]
> Hi, many of our examples' pretrained weights are publicly available. Please see the README of those examples, e.g. [Imagenet](https://github.com/google/flax/tree/main/examples/imagenet), [LM1B](https://github.com/google/flax/tree/main/examples/lm1b), [WMT](https://github.com/google/flax/tree/main/examples/wmt). The Imagenet example also contains a Colab showing how the pretrained weights are loaded.

Thank you, it seems currently the models are still limited, for imagenet we only have RestNet50, no models like Googlenet, Vgg etc. 

And maybe it is better to separate the zoo/hub from examples. I really not expect to find the zoo/hub in examples, as examples are more about teaching people how to use flax, which is different with a zoo/hub which is aimed to provide the pretrained weights to save time and resources. For the purpose of examples, maybe one or two model is enough, but for a zoo, we need more models to support various applications.

So for the long term, is there any plan to build a hub/zoo like tensorflow or pytorch? Or is there any plan to add more examples with pretrained models in examples?

**Reply by marcvanzee** [reply]
We expect that this will happen, but it will mostly be driven by the community, and not by the core maintainers of Flax. We aim to keep our repository manageable, with a few examples that we can maintain. We currently don't have to manpower to also support a model zoo.

HuggingFace currently has over 5000 pretrained models in Flax/JAX, so you may want to take a look there: https://huggingface.co/models?library=jax&sort=downloads
